rcl/src/rcl/event.c

```diff
@@ -79,10 +69,29 @@ rcl_publisher_event_init(
       RCL_SET_ERROR_MSG("Event type for publisher not supported");
       return RCL_RET_INVALID_ARGUMENT;
   }
-  return rmw_publisher_event_init(
+
+  // Allocate space for the implementation struct.
+  event->impl = (rcl_event_impl_t *) allocator->allocate(
+    sizeof(rcl_event_impl_t), allocator->state);
+  RCL_CHECK_FOR_NULL_WITH_MSG(
+    event->impl, "allocating memory failed", ret = RCL_RET_BAD_ALLOC; return ret);
+
+  event->impl->rmw_handle = rmw_get_zero_initialized_event();
+  event->impl->allocator = *allocator;
+
+  ret = rmw_publisher_event_init(
     &event->impl->rmw_handle,
     publisher->impl->rmw_handle,
     rmw_event_type);
+  if (ret != RMW_RET_OK) {
+    goto fail;
+  }
+
+  return RCL_RET_OK;
+fail:
+  allocator->deallocate(event->impl, allocator->state);
+  event->impl = NULL;
+  return ret;
```

Comment:
```suggestion
  return rcl_convert_rmw_ret_to_rcl_ret(ret);
```

Reply:
@clalancette 

Thank you. I'll update it.